contracts/jobs/JobsManager.sol

```diff
@@ -238,6 +236,7 @@ contract JobsManager is ManagerProxyTarget, IVerifiable, IJobsManager {
                 segmentRange: _segmentRange,
                 claimRoot: _claimRoot,
                 claimBlock: block.number,
+                blockHash: block.blockhash(block.number - 1),
```

Comment:
Since the previous code before this PR had the problem of using block.blockhash to reference the claim block when shouldVerifySegment is called it was possible for shouldVerifySegment to fail since block.blockhash only works for up to 256 blocks into the past.

Now we're storing a block hash when the claim is made to be used later. The block hash has to be in the past because the block hash of the current block during a claim is not known until the block is actually mined

Reply:
Ok, this is definitely something we'll have to address before launch. We need to use a future value. I'm thinking an oracle service that submits blockhashes for block n would be ok, or restricting the verificationPeriod to 250 blocks, which would mean that the block + 1 hash would be available on chain, and it could be written down during the verify call.